pennylane/registers.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module contains the :func:`registers` function to build the registers for a given dictionary of registers
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict, _start_wire_index=0):
+    """Returns the registers for the given dictionary of registers. The registers are a dictionary
+    of Wires objects where the key is the register name and the value is the Wires object. The
+    ordering of the Wires objects in the dictionary is based on appearance order first, then on
+    nestedness.
```

Comment:
Do we need to talk about dictionary ordering here? :thinking:

Reply:
```suggestion
    individual wire labels are integers, where the ordering is based on appearance first, then on
    nestedness.
```

@trbromley I think this information should be somewhere in the docs. I think right here is fine!